tests/django/toystore/test_given_models.py

```diff
@@ -157,3 +158,10 @@ def test_constructs_valid_instance(self, instance):
                          instance.choice_field_grouped.lower())
         self.assertEqual(instance.even_number_field % 2, 0)
         self.assertTrue(instance.non_blank_text_field)
+
+
+class TestValidatorInference(TestCase):
+
+    @given(models(User))
+    def test_user_issue_1112_regression(self, user):
+        assert user.username
```

Comment:
I'm surprised this actually passes. When I tried the test locally with an explicit strategy, it seemed to pass with an empty user name.

I have a suspicion that this means that `RegexValidator` might not be applied when `blank` is set to `True` or something like that. Another possibility might be that we don't run validation when a field is set explicitly.

It's only a problem with this PR if it's the former, but either way would you mind investigating? (If it's the latter don't worry about fixing this in this PR but we should probably open an issue)

Reply:
It's the latter - we only use the validators to help us infer a correct strategy.

I'm not sure permitting explicit strategies which would fail validation should be considered a bug.  In any case, adding a filter to user-supplied strategies would be a new feature and would almost certainly break user code in the wild :sob: